test/core/security/grpc_authorization_policy_provider_test.cc

```diff
@@ -164,6 +186,24 @@ TEST(AuthorizationPolicyProviderTest, FileWatcherRecoversFromFailure) {
   auto provider = FileWatcherAuthorizationPolicyProvider::Create(
       tmp_authz_policy->name(), /*refresh_interval_sec=*/1);
   ASSERT_TRUE(provider.ok());
+  absl::Notification on_first_reload_done;
+  absl::Notification on_second_reload_done;
+  bool first_reload = true;
+  std::function<void(grpc_status_code code, const char* error_details)>
+      callback = [&on_first_reload_done, &on_second_reload_done, &first_reload](
```

Comment:
As mentioned elsewhere, instead of having one callback deal with all of this state, let's just reset the callback after the first reload is done.

Reply:
Done.